src/app/components/wrapper-components/online-game-wrapper/online-game-wrapper.component.ts

```diff
@@ -130,19 +130,25 @@ export class OnlineGameWrapperComponent extends GameWrapper implements OnInit, O
     private isOpponent(player: PlayerOrNone): boolean {
         return this.observerRole !== player.value;
     }
-    private async redirectIfPartIsInvalid(): Promise<void> {
-        const gameType: string = this.extractGameTypeFromURL();
-        const partValidity: MGPValidation =
-            await this.gameService.getPartValidity(this.currentPartId, gameType);
-        if (partValidity.isFailure()) {
-            // note, option if WRONG_GAME_TYPE to redirect to another page
+    private async redirectIfPartOrGameIsInvalid(): Promise<void> {
+        const gameName: string = this.extractGameNameFromURL();
+        const gameExists: boolean = GameInfo.ALL_GAMES().some((gameInfo: GameInfo) => gameInfo.urlName === gameName);
+        if (gameExists === false) {
             this.routerEventsSub.unsubscribe();
-            await this.router.navigate(['/notFound', GameWrapperMessages.NO_MATCHING_PART()], { skipLocationChange: true } );
+            await this.router.navigate(['/notFound', GameWrapperMessages.NO_MATCHING_GAME(gameName)], { skipLocationChange: true } );
+        } else {
+            const partValidity: MGPValidation =
+                await this.gameService.getPartValidity(this.currentPartId, gameName);
+            if (partValidity.isFailure()) {
+                // note, option if WRONG_GAME_TYPE to redirect to another page
```

Comment:
that's rather unclear, what ?

Reply:
le commentaire ? Yep j'l'ai pas changé mais je proposerais bien de le supprimer ! Je pense que l'idée est que si tu fais /P5/partie-existante, on redirige sur /P4/partie-existante, mais ça pue du cul : ça n'arriverait jamais en pratique (ou alors tu mérite la 404 !), et ça rajoute de la logique pour un cas inutile (trouver le type de jeu correspondant)

Reply:
/unexisting-component/** => unexisting-component n'existe pas, tu t'es gourré mr user
/existing-component/unexisting-game/** the game unexisting-game don't exist, disoulé missiou.
/existing-component/existing-game/unexisting-room this room has no game linked to it, you might have done a typo

On check de gauche à droite dans l'url, ça me semble clair ? Aucunes redirection (au yeux de l'url en tout cas)
ça fait sens ?

Reply:
Yep!

Reply:
Donc l'idée de ce commentaire c'est quoi, j'pige toujours rien à cette phrase :D !

Reply:
j'sais pas, c'pas mon commentaire :D Y'en a plein des comme ça où je comprends pas, c'est le cas pour 90% des TODO qu'il reste !

Reply:
faisons simple, tu peux supprimer :p

Reply:
poubellé !